### PR TITLE
Limit the ethCall gas to 15_000_000, otherwise the call will fail

### PR DESCRIPTION
**Pull request by dimitrovmaksim**
Signed-off-by: Maksim Dimitrov <dimitrov.maksim@gmail.com>

**Description**:
I'm not sure if this is the best way to do this, but ethCalls with gas over 15_000_000 will fail with BUSY error, because the Hedera limit is set to 15_000_000. This breaks the integration with TheGraph, when executing `eth_call`s from the event handlers, because the graph-node reserves all 50_000_000 gas.

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-json-rpc-relay/issues/511

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -792,7 +792,13 @@ export class EthImpl implements Eth {
           gas = call.gas;
         }
       }
-      
+
+      // Gas limit for `eth_call` is 50_000_000, but the Hedera network limit is 15_000_000
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:796]: Assuming it's safe to cap this we can leave it here for now.
Please add a trace log.
Also add a test  to verify the submitted transaction has the gas capped.

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
Missing log and test

File: packages/relay/src/lib/eth.ts [hunk 2]
```diff
@@ -792,7 +792,14 @@ export class EthImpl implements Eth {
...
+      // With values over 15_000_000 the call will fail with BUSY error
+      if (gas > 15_000_000) {
+        this.logger.trace(`${requestIdPrefix} eth_call gas amount (${gas}) exceeds network limit, capping gas to 15_000_000`);
+        gas = 15_000_000;
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:800]: nit: could have made `15_000_000` a const
Can make this change in a future PR, no biggie

**Review by Nana-EC — APPROVED** [review]
LG

File: packages/relay/src/lib/eth.ts (hunk 2 above)
Comment (dimitrovmaksim) [reply packages/relay/src/lib/eth.ts:800]: I'll still probably have to merge main at some point to fix the dapp tests, I can push that change too

**Review by Nana-EC — APPROVED** [review]

**Comment by Nana-EC** [comment]
@dimitrovmaksim, all your commits need to be signed

**Comment by dimitrovmaksim** [comment]
> @dimitrovmaksim, all your commits need to be signed

All of them seem to have the "Signed-off-by:" message

**Comment by dimitrovmaksim** [comment]
Closing in favor of https://github.com/hashgraph/hedera-json-rpc-relay/pull/526

**Pull request closed without merging**